src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1061,6 +1061,39 @@ static bool address_available(sd_dhcp_server *server, be32_t address) {
         return true;
 }
 
+static int server_get_static_lease(sd_dhcp_server *server, const DHCPRequest *req, DHCPLease **ret) {
+        DHCPLease *static_lease;
+        _cleanup_free_ uint8_t *data = NULL;
+
+        static_lease = hashmap_get(server->static_leases_by_client_id, &req->client_id);
+        if (static_lease) {
+                *ret = TAKE_PTR(static_lease);
```

Comment:
`static_lease` here in this function does not have `_cleanup_` attribute, then `TAKE_PTR()` is not necessary.